lib/taskana-core-test/src/test/java/acceptance/task/requestchanges/RequestChangesWithSpiAccTest.java

```diff
@@ -0,0 +1,267 @@
+        .buildAndStore(workbasketService);
+
+    defaultObjectReference = DefaultTestEntities.defaultTestObjectReference().build();
+  }
+
+  private TaskBuilder createTaskInReviewByUser(String owner) {
+    return createDefaultTask().owner(owner).state(TaskState.IN_REVIEW);
+  }
+
+  private TaskBuilder createDefaultTask() {
+    return TaskBuilder.newTask()
+        .classificationSummary(defaultClassificationSummary)
+        .workbasketSummary(defaultWorkbasketSummary)
+        .primaryObjRef(defaultObjectReference);
+  }
+
+  static class ExceptionThrower implements AfterRequestChangesProvider {
+
+    @Override
+    public void initialize(TaskanaEngine taskanaEngine) {}
+
+    @Override
+    public Task afterRequestChanges(Task task) {
+      throw new SystemException("I AM THE EXCEPTION THROWER (*_*)");
+    }
+  }
+
+  static class TaskModifierAndTransferrer implements AfterRequestChangesProvider {
+
+    private TaskanaEngine taskanaEngine;
+
+    @Override
+    public void initialize(TaskanaEngine taskanaEngine) {
+      this.taskanaEngine = taskanaEngine;
+    }
+
+    @Override
+    public Task afterRequestChanges(Task task) throws Exception {
+      task.setCustomField(TaskCustomField.CUSTOM_3, "bla");
+      task = taskanaEngine.getTaskService().updateTask(task);
+      task = taskanaEngine.getTaskService().transfer(task.getId(), NEW_WORKBASKET_KEY, "DOMAIN_A");
+      return task;
+    }
+  }
+
+  static class ClassificationUpdater implements AfterRequestChangesProvider {
+
+    public static final String SPI_CLASSIFICATION_NAME = "Neuer Classification Name";
+
+    private TaskanaEngine taskanaEngine;
+
+    @Override
+    public void initialize(TaskanaEngine taskanaEngine) {
+      this.taskanaEngine = taskanaEngine;
+    }
+
+    @Override
+    public Task afterRequestChanges(Task task) throws Exception {
+      Classification newClassification =
+          defaultTestClassification().buildAndStore(taskanaEngine.getClassificationService());
+
+      task.setClassificationKey(newClassification.getKey());
+      task = taskanaEngine.getTaskService().updateTask(task);
+
+      newClassification.setName(SPI_CLASSIFICATION_NAME);
+      taskanaEngine.getClassificationService().updateClassification(newClassification);
+
+      return task;
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  @WithServiceProvider(
+      serviceProviderInterface = AfterRequestChangesProvider.class,
+      serviceProviders = TaskModifierAndTransferrer.class)
+  class SpiModifiesTask {
+
+    @TaskanaInject TaskService taskService;
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ReturnModifiedTask_When_SpiModifiesAndTransfersTask() throws Exception {
+      Task task = createTaskInReviewByUser("user-1-1").buildAndStore(taskService);
+
+      Task result = taskService.requestChanges(task.getId());
+
+      assertThat(result.getCustomField(TaskCustomField.CUSTOM_3)).isEqualTo("bla");
```

Comment:
Should we make "bla" a constant?

We didn't do this in the other test aswell.

Reply:
I will create the constant in both tests